docker-compose-rule-core/src/main/java/com/palantir/docker/compose/connection/waiting/ClusterHealthCheck.java

```diff
@@ -52,13 +56,18 @@ static ClusterHealthCheck nativeHealthChecks() {
         return cluster -> {
             Set<String> unhealthyContainers = new LinkedHashSet<>();
             try {
+                log.info("Checking health of containers {}",
```

Comment:
these may be a little spammy, FYI. Think this is done in a fairly tight loop, so you may end up with many log lines. 

Instead, I'd recommend doing this like:

```java
Set<String> lastUnhealthyContainers = new LinkedHashSet<>();
return new ClusterHealthCheck() {
    @Override
    public SuccessOrFailure isClusterHealthy(Cluster cluster) throws InterruptedException {
        Set<String> currentUnhealthyContainers = new LinkedHashSet<>();
        boolean healthyContainerSetChanged = false;
        for (Container container : cluster.allContainers()) {
            State state = container.state();
            if (state == State.UNHEALTHY) {
                currentUnhealthyContainers.add(container.getContainerName());
            }
        }

        if (currentUnhealthyContainers.equals(lastUnhealthyContainers)) {
              log(currentUnhealthyContainers);
              lastUnhealthyContainers.clear();
              lastUnhealthyContainers.addAll(currentUnhealthyContainers);
        }

        if (!currentUnhealthyContainers.isEmpty()) {
            return SuccessOrFailure.failure("The following containers are not healthy: " ...);
        }

        return SuccessOrFailure.success();
    }
}

```

Reply:
where you only log if the state has actually changed.